orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/kubernetes/KubernetesContainerFinder.groovy

```diff
@@ -63,6 +63,15 @@ class KubernetesContainerFinder {
 
     containers.forEach { container ->
       if (container.imageDescription.fromContext) {
+	//  If there are parallel Find images stages , then deploymentDetails would be fetched from it. 
```

Comment:
I believe the correct fix for this is to support (or favor) using the non-global context when looking for `amiDetails`. 

In the short-term, I suspect you can work around this by adding a small wait stage (~30s) to one of your branches. That should add enough buffer for the first deploy to start before the second `Find Image` overwrites the global context.